### ISSUE TITLE
Adding emacs-macport

### ISSUE THREAD

**Issue by knl** [nix-community/emacs-overlay#214]
As this overlay tracks various emacs packages, I'm wondering would it make sense to add emacs-macport to it as well? If yes, I'm willing to add it.

**Comment by jhenahan** [comment]
I actually used the overlay with the Mac port for a while just by specifying it as the `package` as you would normally for one of the overlay versions (since it's tracked in nixpkgs). I expect the hard part will be getting CI working, but then again I certainly wouldn't hate having cached builds for more Darwin stuff.

**Comment by montchr** [comment]
> I actually used the overlay with the Mac port for a while just by specifying it as the `package` as you would normally for one of the overlay versions (since it's tracked in nixpkgs).

@jhenahan I didn't realize this was a thing. But re: "for a while": I'm curious why you stopped using it? Considering that it looks like `emacs-mac` (the source repo name on Bitbucket) [now has native-comp support](https://github.com/railwaycat/homebrew-emacsmacport/issues/274#issue-1100744030) **and** it comes with the `no-titlebar` patch enabled by default, this is an interesting option to me. 

> I certainly wouldn't hate having cached builds for more Darwin stuff.

I certainly wouldn't hate that either. I would actually appreciate that quite a bit 😄 

All that said, the version in nixpkgs (even the `master` branch) has not been updated to include support for native-comp / 28.0.92, but the source repo appears to add support for that version. https://bitbucket.org/mituharu/emacs-mac/src/work/

Considering that mituharu/railwaycat's repo is effectively a "fork" of Emacs with macOS-oriented patches applied, and the nixpkgs version is months out of date, I can totally see how `emacs-mac` would fit within the scope of `emacs-overlay`.

> As this overlay tracks various emacs packages, I'm wondering would it make sense to add emacs-macport to it as well? If yes, I'm willing to add it.

@knl FWIW you've got my vote. I'm by no means involved in this project but let me know if you need a volunteer to test.

**Comment by montchr** [comment]
Related recent-ish updates regarding the version on nixpkgs:

- https://github.com/NixOS/nixpkgs/issues/127902#issuecomment-1055676905
- https://github.com/NixOS/nixpkgs/pull/155360

**Comment by Atemu** [comment]
FYI, the emacsMacport integration PR linked above has been merged which means it does include native-comp now and can be used more easily for the purposes of the overlay.

You only need to pass a source and macportVersion.

**Comment by dhess** [comment]
Has anyone been able to get this to work on `aarch64-darwin`? Mine always dies trying to build something `clang`- or `llvm`-related.

**Comment by Atemu** [comment]
This seems to be an issue on the Nixpkgs side though as other ports of this package work just fine.

**Comment by dhess** [comment]
Yes, it's a nixpkgs issue, I was just curious whether anyone else had seen the issue, since you mentioned emacsMacport.
